components/function-controller/design/subresources_templates_proposal.md

```diff
@@ -0,0 +1,84 @@
+# Function Subresources Templating for Serverless v1alpha2
+
+## Summery
+
+The current Serverless API allows for limited ability to configure the generated functions subresources (deployment, build job). Additionally, it doesn't allow for separate configuration for build and functions resources except for memory/cpu resources.
+
+This proposes changes to the current v1alpha2 Serverless API to support templating of function sub resources _without_ the need to rollout a new API version.
+
+## Motivation
+
+Give Serverless users the ability to:
+- Separate configuration for function deployments and build jobs.
+- Configure volume mounts for function subresources
+- Configure sidecars for function deployments (?)
+
+### Goals
+
+- Add more flexibility to the Serverless API.
+- Provide a rollout plan to implement this functionality in small increments and avoid the need to rollout a new API version
+
+### Non-Goals
+- Define a new Spec API version
+
+## Proposal
+
+Use a modified version of [PodTemplateSpec](https://github.com/kubernetes/kubernetes/blob/64ed9145452d2d1d324d2437566f1ea1ce76f226/pkg/apis/core/types.go#L3443) as a base for function build/deployment templates. 
+
+A modified version is needed because we need to protect certain parts of the Pod spec (commands and args for the main container in the pod for example).
```

Comment:
```suggestion
A modified version is needed because we need to protect certain parts of the Pod specification, for example, commands and arguments for the main container in the Pod.
```